tally_ho/apps/tally/views/audit.py

```diff
@@ -45,6 +47,12 @@ def audit_action(audit, post_data, result_form, url):
         if audit.resolution_recommendation ==\
                 AuditResolution.MAKE_AVAILABLE_FOR_ARCHIVE:
             audit.for_superadmin = True
+        elif audit.action_prior_to_recommendation ==\
+                ActionsPrior.REQUEST_AUDIT_ACTION_FROM_FIELD or\
+                ActionsPrior.REQUEST_COPY_FROM_FIELD:
```

Comment:
This line has no effect, since `ActionsPrior.REQUEST_COPY_FROM_FIELD` is 0 it's boolean value is false so this is equivalent to the proposition `... or False` which is equivalent to `...`

Do you want to test if `audit.action_prior_to_recommendation == ActionsPrior.REQUEST_COPY_FROM_FIELD`? If so, you have to either add the equality to the 2nd disjunct,

```python
audit.action_prior_to_recommendation == ActionsPrior.REQUEST_AUDIT_ACTION_FROM_FIELD or\
audit.action_prior_to_recommendation == ActionsPrior.REQUEST_COPY_FROM_FIELD
```
or, and probably better, use `in`, which is equivalent to the above:

```python
audit.action_prior_to_recommendation in [
    ActionsPrior.REQUEST_AUDIT_ACTION_FROM_FIELD, ActionsPrior.REQUEST_COPY_FROM_FIELD]
```

Either way, some tests should validate that this conditional is correct, i.e. you should write a test that the current code fails because of the error in the code as currently written.

Reply:
cool fix this.